lib/kubernetes-deploy/kubernetes_resource/job.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class Job < KubernetesResource
+    TIMEOUT = 10.minutes
+
+    def deploy_succeeded?
+      # Don't block deploys for long running jobs,
+      # Instead report success when there is at least 1 active
+      return false unless deploy_started?
+      done? || running?
+    end
+
+    def deploy_failed?
+      return false unless deploy_started?
+      return false unless @instance_data.dig("spec", "backoffLimit").present?
+      (@instance_data.dig("status", "failed") || 0) >= @instance_data.dig("spec", "backoffLimit")
+    end
+
+    def status
+      if !exists?
+        super
+      elsif done?
+        "Succeeded"
+      elsif running?
+        "Started"
+      elsif deploy_failed?
+        "Failed"
+      else
+        "Unknown"
+      end
+    end
+
+    private
+
+    def done?
+      (@instance_data.dig("status", "succeeded") || 0) == @instance_data.dig("spec", "completions")
+    end
+
+    def running?
+      now = Time.now.utc
+      start_time = @instance_data.dig("status", "startTime")
+      # Wait 5 seconds to ensure job doesn't immediately fail.
```

Comment:
I'm not sure I understand this; `deploy_failed?` doesn't look at `running?`, so afaict the effect of this is to potentially delay declaring success.

Reply:
This actually prevents declaring success before k8s has tried to run the command. `test_jobs_can_fail` which uses the fake command `/not/a/command` will actually report as running for a brief period (& therefore successful) if we don't introduce a slight delay.

Reply:
Oh, I see. Requiring it to run for at least 5 seconds seems reasonable. I wonder if adding a readiness probe to the container would work too, but this will make our tool work better in more cases regardless.